### PR TITLE
Add action intg test

### PR DESCRIPTION
**Pull request by jaegukhyun**
### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by harimkang — APPROVED** [review]

File: src/otx/engine/hpo/hpo_api.py [hunk 1]
```diff
@@ -177,7 +178,7 @@ def _get_default_search_space(self) -> dict[str, Any]:
         if isinstance(self._engine.optimizer, list):
             for i, optimizer in enumerate(self._engine.optimizer):
                 search_space[f"optimizer.{i}.keywords.lr"] = self._make_lr_search_space(optimizer)
-        elif isinstance(self._engine.optimizer, OptimizerCallable):
+        elif self._engine.optimizer is not None:
```
Comment (sungmanc) [inline src/otx/engine/hpo/hpo_api.py:181]: Why this change is needed? Is there any case the optimizer isn't `OptimizerCallable`?

Comment (jaegukhyun) [reply src/otx/engine/hpo/hpo_api.py:181]: I'm not sure, @eunwoosh Could you explain this?

**Review by sungmanc — APPROVED** [review]
Although I left a comment, I think that's not crucial for now. So, I approved

File: src/otx/engine/hpo/hpo_api.py (hunk 1 above)
Comment (eunwoosh) [reply src/otx/engine/hpo/hpo_api.py:181]: No, but `OptimizerCallable` can't be used for `isinstance` because it just a `Callable` instance.
